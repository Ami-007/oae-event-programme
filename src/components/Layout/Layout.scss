@use "../../assets/sass/typography" as *;
@use "../../assets/sass/variables" as *;
:root {
  --theme-color-primary: #ffffff;
  --theme-color-text: #000000;
  --theme-color-highlight: #000000;
  --font-size: 100%;

  font-size: var(--font-size);
}

.layout {
  // margin-bottom compensates for the height of the fixed navbar
  padding-bottom: 6rem;
  background-color: $background-color;
  min-height: 100vh;
}
