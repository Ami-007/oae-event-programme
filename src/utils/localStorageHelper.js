export const getDarkTheme = () => {
  //   if (darkTheme == true) {
  //     localStorage.setItem("--background-color", darkTheme);
  //   } else {
  //     localStorage.setItem("theme", "light");
  //   }
  console.log(localStorage.getItem("darkMode"));

  // convert isDarkMode to a Boolean
  // return that boolean

  return true;
};

export const setDarkTheme = (isDarkMode) => {
  localStorage.setItem("darkMode", isDarkMode.toString());
};
