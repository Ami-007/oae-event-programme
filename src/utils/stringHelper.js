export const getFontType = (fontType) => {
  if (typeof fontType === "string") {
    fontType = parseInt(fontType);
  }
  if (fontType === 1) {
    return "modern";
  } else if (fontType === 2) {
    return "classic";
  } else {
    return "modern";
  }
};

export default getFontType;
