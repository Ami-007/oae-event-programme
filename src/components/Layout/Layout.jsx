import { useEffect } from "react";
import { getDarkTheme } from "../../utils/localStorageHelper";
import Navbar from "../Navbar/Navbar";
import "./Layout.scss";

const Layout = (props) => {
  const {
    theme = "modern",
    children,
    eventId,
    themeColorPrimary,
    themeColorText,
    themeColorHighlight,
  } = props;

  const darkTheme = getDarkTheme();

  useEffect(() => {
    let { style } = document.documentElement;

    // set theme colours
    themeColorPrimary &&
      style.setProperty("--theme-color-primary", themeColorPrimary);
    themeColorText && style.setProperty("--theme-color-text", themeColorText);
    themeColorHighlight &&
      style.setProperty("--theme-color-highlight", themeColorHighlight);

    //set dark/ light mode colours
    style.setProperty("--background-color", darkTheme ? "#333333" : "#fafafc");
    style.setProperty("--text-color", darkTheme ? "#e4e5f0" : "#333333");
    style.setProperty("--header-color", darkTheme ? "#ffffff" : "#333333");
    style.setProperty("--sub-header-color", darkTheme ? "#bfbfc7" : "#333333");
    style.setProperty("--hr-color", darkTheme ? "#bfbfc7" : "#e4e5f0");
  }, [themeColorPrimary, themeColorText, themeColorHighlight]);

  return (
    <div className={`layout ${theme}`}>
      <div>{children}</div>
      <Navbar eventId={eventId} />
    </div>
  );
};

export default Layout;
