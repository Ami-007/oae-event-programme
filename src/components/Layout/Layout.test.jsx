import Layout from "./Layout";
import { customRender } from "../../utils/testUtils";

test("Renders Layout component with default page", () => {
  const { container } = customRender(
    <Layout eventId="123">
      <p>sample children</p>
    </Layout>
  );
  expect(container).toMatchSnapshot();
});
