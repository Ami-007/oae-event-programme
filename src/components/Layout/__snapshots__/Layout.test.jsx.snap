// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Layout component with default page 1`] = `
<div>
  <div
    class="layout"
  >
    <div>
      <p>
        sample children
      </p>
    </div>
    <nav
      class="navbar"
    >
      <div
        class="navbar__content"
      >
        <a
          class="navbar__schedule-link"
          href="/123/event-schedule"
        >
          <img
            alt="schedule icon"
            class="navbar__icon"
            src="schedule.svg"
          />
        </a>
        <span
          class="navbar__title"
        >
          SCHEDULE
        </span>
      </div>
      <div
        class="navbar__content"
      >
        <a
          class="navbar__home-link"
          href="/123"
        >
          <img
            alt="home icon"
            class="navbar__icon"
            src="home.svg"
          />
        </a>
        <span
          class="navbar__title"
        >
          HOME
        </span>
      </div>
      <div
        class="navbar__content"
      >
        <a
          class="navbar__about-link"
          href="/123/our-story"
        >
          <img
            alt="about icon"
            class="navbar__icon"
            src="info.svg"
          />
        </a>
        <span
          class="navbar__title"
        >
          ABOUT
        </span>
      </div>
    </nav>
  </div>
</div>
`;
